libraries/AP_MSP/AP_MSP_Telem_Backend.cpp

```diff
@@ -577,6 +577,19 @@ void AP_MSP_Telem_Backend::msp_handle_airspeed(const MSP::msp_airspeed_data_mess
 #endif
 }
 
+uint32_t AP_MSP_Telem_Backend::get_osd_flight_mode_bitmask(void)
+{
+    uint32_t mode_mask = 0;
+    const AP_Notify& notify = AP::notify();
+
+    // set arming status
+    // Note: BOXARM bit is the same for DJI,BF and INAV
+    if (notify.flags.armed) {
```

Comment:
```suggestion
    // Note: BOXARM bit is the same for DJI,BF and INAV
    if (hal.util->get_soft_armed()) {
```

Reply:
ok, fixed and removed the unneeded mode_mask variable as well